### ISSUE TITLE
"missing whitespace" warning is reported even with every whitespace rule turned off

### ISSUE THREAD

**Issue by chriszrc** [palantir/tslint#4451]
### Bug Report

- __TSLint version__: 5.12.0
- __TypeScript version__: 3.1.6
- __Running TSLint via__:  Vue CLI 

#### TypeScript code being linted

```ts
beforeDestroy(){
               ^
  window.removeEventListener('resize', this.handleWindowResize);
}
```

with `tslint.json` configuration:

```json
{
  "defaultSeverity": "warning",
  "extends": [
    "tslint:recommended"
  ],
  "linterOptions": {
    "exclude": [
      "node_modules/**",
      "**/*.json"
    ]
  },
  "rules": {
    "quotemark": false,
    "indent": [true, "spaces", 2],
    "interface-name": false,
    "ordered-imports": false,
    "object-literal-sort-keys": false,
    "no-consecutive-blank-lines": false,
    "whitespace": false,
    "typedef-whitespace": false,
    "no-trailing-whitespace": false,
    "no-irregular-whitespace": false,
    "space-before-function-paren": false,
    "space-within-parens":false,
    "prefer-const":false,
    "max-line-length":false,
    "comment-format":false,
    "member-access":false,
    "arrow-parens":false
  }
}
```

#### Actual behavior
I still get the whitespace error, I'm assuming because of the "check-preblock" check in the whitespace rule. All the other rules are correctly being turned off by the configuration, but not this one

#### Expected behavior
No more whitespace errors

**Comment by chriszrc** [comment]
I also tried 5.11.0 too, same problem. Would there be a better candidate version to try and use?

**Comment by FDIM** [comment]
I also get a bit strange warning for a question mark with 5.11.0

```ts
@Output()` onRemove? = new EventEmitter<void>();
                   ^---- missing whitespace
````

It goes away if I set the type as below or If I disable "check-preblock" rule
```
@Output() onRemove?: EventEmitter<void> = new EventEmitter<void>();

```

**Comment by chriszrc** [comment]
I'm new to tslint, so I'm just checking here, should setting the entire whitespace rule to false like this:

```
"whitespace": false,
```

disable every whitespace check? Or do I need to list out the individual rules to disable them all?

**Comment by FDIM** [comment]
Looks like the error I described before was introduced with `5.9.0` - so @chriszrc I would downgrade to `5.8.0` if you do not rely on new features

**Comment by adidahiya** [comment]
If you want to see which rules are causing lint failures (so you can disable them), try using the "verbose" or "stylish" formatter. There are open issues to make this formatter the default (#3460) and group all formatting rules together (#3592).

**Comment by chriszrc** [comment]
+1 for it being the default, but actually I tried to change the formatter, but without luck, since it appears you can't change it from tslint.json, and I'm running tslint from the vue cli, which runs it from a webpack plugin that appears to not respect additional flags passed to it from the command line. And I haven't worked out how to run tslint from the command line, because it's a vue project, and the .vue files would need to be preprocessed first :( 

But more importantly, are you saying that my tslint.json from the beginning of this issue could be missing rules or hasn't properly disabled some of the whitespace rules?

**Comment by adidahiya** [comment]
@chriszrc I'm not sure; it seems like you're correctly disabled all the right rules but I'm still seeing the lint failures reported in the [tslint playground](https://palantir.github.io/tslint-playground/?saved=N4Igxg9gJgpiBcIBmBXAdmALgSwmgBAEYxIQBOMAIjAM6ZkQCeAFAJTAA6B+A7tmlAg8AdBQC2EAG4wAotLSYAMtjow0MMvmYByCjWwAvGNoA0+ABYBDAQBsYAdX6CeAJVqGYrANxcAvlxATcDwkbABzBBBObg4QWCRLFBtMAGUYaTJsTEZY+HxYnksyNH4w2JMufHyQGAAPTDUoGlz8AG1KqurMGht+THgKSDExRpgoWI6AXQqYkF6FDQB5AAccPGaEfGjO6rqwGxRYFvbuHdi0aBgAfQkoJNoAegAqJ-KOs5AX5+EAKxo8CanfCTDq+GZVWJke4bPLbTqxACOKAgDTERQA1i0EjYaDBwfCQE41JhjvQUHjqjRlpYwLRyvgAEzTd7VPoaBK0gC0aEsIyxlhxeJZsXIsAoUE52DEy3I3X5gvxEJAEEIPxgWE5vQaZAFnP+ZEwnPRMEYMPw2NxiuqF05kDQuLAKBw0k5hBs1nRmv4dM2FqFQIK5iytGptPlluFIGyyzGJE5PCDDSpNJg4f9Hxt9Es2HmYXjiZDKbTVvOEElZAoYSSRXzweTYd9AojAZA9ZgrpI5HbqAwazQnOpFDQLTJ6YJbfjWSD-cHahhfpLIGWFCQGlt6xJCAXkbRtS96k1ajCmHMuW3LaGIwUnNIZDRm-g54+IzExDInJptJo86bY6VRQYHgByKOcz1-UE-BAXwgA) so this looks like a regression, but I haven't had time to dig into it yet:

![image](https://user-images.githubusercontent.com/723999/51202860-b2e19000-18cd-11e9-973b-f91b32c6b731.png)

**Comment by chriszrc** [comment]
@adidahiya ok, thanks for confirming. Not sure if this is just due to the formatter or not, but the warning I get locally is `missing whitespace` not the full `Missing whitespace before function parens`. Thanks for looking into this-

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is being deprecated** and no longer accepting pull requests for major new changes or features. See #4534. 😱

If you'd like to see this change implemented, you have two choices:
* **Recommended:** Check if this is available in [ESLint](https://eslint.org) + [typescript-eslint](https://typescript-eslint.io) ✅
* _Not Recommended: Fork TSLint locally_ 🤷‍♂️

👋 It was a pleasure open sourcing with you!

_If you believe this message was posted here in error, please comment so we can re-open the issue!_

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
